include/z64game_over.h

```diff
@@ -0,0 +1,30 @@
+#ifndef Z64GAME_OVER_H
+#define Z64GAME_OVER_H
+
+#include "ultra64.h"
+
+struct PlayState;
+
+
+typedef enum GameOverState {
+    /*  0 */ GAMEOVER_INACTIVE,
+    /*  1 */ GAMEOVER_DEATH_START,
+    /*  2 */ GAMEOVER_DEATH_WAIT_GROUND, // wait for player to fall and hit the ground
+    /*  3 */ GAMEOVER_DEATH_FADE_OUT, // wait before fading out
+    /* 20 */ GAMEOVER_REVIVE_START = 20,
+    /* 21 */ GAMEOVER_REVIVE_RUMBLE,
+    /* 22 */ GAMEOVER_REVIVE_WAIT_GROUND, // wait for player to fall and hit the ground
+    /* 23 */ GAMEOVER_REVIVE_WAIT_FAIRY, // wait for the fairy to rise all the way up out of player's body
+    /* 24 */ GAMEOVER_REVIVE_FADE_OUT // fade out the game over lights as player is revived and gets back up
+} GameOverState;
+
+typedef struct GameOverContext {
+    /* 0x0 */ u16 state;
+} GameOverContext; // size = 0x2
```

Comment:
This seems odd, does it really have to be its own struct to match? Was it used for more in OOT?

Reply:
It is used mostly here
<https://github.com/zeldaret/mm/blob/f3d2c56d1deed7194c77909ee41ed5b8ef3ef8a2/src/code/z_game_over.c#L10>
Those uses makes us think it must be a struct